src/js/components/Institutions.jsx

```diff
@@ -1,297 +1,101 @@
-                <h1>{this.props.filingPeriod} filing period</h1>
-                <p>The filing period is open. You may file HMDA data for your authorized institutions below.</p>
-                <p>Your progress will be saved if you leave the platform before completing your filing.</p>
-              </header>
-              :
-              <Alert
-                type="warning"
-                heading='The filing period is closed.'>
-                <p>The platform remains available outside of the filing period to upload, test, and validate HMDA data.</p>
-              </Alert>
-            : null
-          }
-          {!this.props.filings.fetched || this.props.filings.isFetching || this.props.submission.isFetching
-            ? <LoadingIcon />
-            : this.props.filings.fetched && this.props.filings.filings.length === 0
-              ? <Alert type="error">
-                  <p>
-                    There is a problem with your filing. Please contact <a href="mailto:hmdahelp@cfpb.gov">HMDA Help</a>.
-                  </p>
-                </Alert>
-              : this.props.filings.filings.map((filingObj, i) => {
-                  const filing = filingObj.filing
-                  const submission = this.props.submission.id &&
-                    this.props.submission.id.institutionId === filing.institutionId
-                    ? this.props.submission
-                    : filingObj.submissions[0]
-                  const status = submission && submission.status
-                  const institution = getInstitutionFromFiling(
-                    institutions,
-                    filing
-                  )
-
-                  if (!institution) return
-                  return (
-                    <div key={i} className="usa-grid-full">
-                      <section className="institution">
-                        <div className="current-status">
-                          <h3>{institution.name} - {institution.id}</h3>
-                          {renderStatus(
-                            institution.id,
-                            filing,
-                            submission,
-                            this.props.onDownloadClick
-                          )}
-
-                          {renderViewButton(
-                            status,
-                            filing.institutionId,
-                            filing.period
-                          )}
-
-                          {renderRefileButton(status, filing)}
-                        </div>
-
-                        {renderPreviousSubmissions(
-                          filingObj.submissions,
-                          this.props.onDownloadClick,
-                          institution.id,
-                          filing.period
-                        )}
-                      </section>
-                    </div>
-                  )
-                })}
+          <InstitutionsHeader filingPeriod={this.props.filingPeriod} />
```

Comment:
The logic in the render here is still a bit opaque. Maybe some helper functions could replace the inline stuff
A `renderBody` fn could call an `isLoading` and a `resolveSubmission` function etc. to mask/separate this logic a bit

Reply:
c387102 does a bit here. I like that the actual renders all happen within `render()` this way, leaving the `.map()` there. We can always do more later.